File: 0359-Logger Rate Limiter/0359-Logger Rate Limiter.cpp
class Logger {
public:
    /** Initialize your data structure here. */
    Logger() {
        
    }
    
    /** Returns true if the message should be printed in the given timestamp, otherwise returns false.
        If this method returns false, the message will not be printed.
        The timestamp is in seconds granularity. */
    bool shouldPrintMessage(int timestamp, string message) {
        if (table.find(message) != table.end() && timestamp - table[message] < 10)
        {
            return false;
        }
        else
        {
            table[message] = timestamp;
            return true;
        }
    }
    
private:
    unordered_map<string, int> table;
};

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * bool param_1 = obj.shouldPrintMessage(timestamp,message);
 */
 

File: 0301-0400/0359-Logger Rate Limiter/0359-Logger Rate Limiter.py
class Logger:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.time = {}


    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:
        """
        Returns true if the message should be printed in the given timestamp, otherwise returns false.
        If this method returns false, the message will not be printed.
        The timestamp is in seconds granularity.
        """
        if message in self.time and timestamp - self.time[message] < 10:
            return False
        else:
            self.time[message] = timestamp
            return True


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)
